opal/corelib/io.rb

```diff
@@ -3,6 +3,25 @@ class IO
   SEEK_CUR = 1
   SEEK_END = 2
 
+  attr_reader :eof
+  attr_reader :lineno
+
+  def initialize
+    @eof = false
+    @lineno = 0
+  end
+
+  def read
+    if @eof
+      ''
+    else
+      res = File.read(@path)
```

Comment:
I'm not sure to understand how pluggable write is working. How `STDOUT` or `STDERR` is chosen ?
But I can still mimic the behavior, something like https://gist.github.com/Mogztter/7f0c7acf4c94f3002c6422ac26572d67  ?